lib/chef/knife/ec2_server_create.rb

```diff
@@ -1536,13 +1536,15 @@ def check_windows_password_available(server_id)
       def windows_password
         if not locate_config_value(:winrm_password)
           if locate_config_value(:identity_file)
-            print "\n#{ui.color("Waiting for Windows Admin password to be available", :magenta)}"
-            print(".") until check_windows_password_available(@server.id) {
-              puts("done")
-            }
-            response = connection.get_password_data(@server.id)
-            data = File.read(locate_config_value(:identity_file))
-            config[:winrm_password] = decrypt_admin_password(response.body["passwordData"], data)
+            if @server
```

Comment:
We need to confirm this do we required this line to set user data for Administrator since ec2 creates default user as Administrator and it generates its own password. So our main problem is here we unnecessarily as far as I understand we don't need to add that net user command for Administrator here https://github.com/MsysTechnologiesllc/knife-ec2/blob/c3371c98ce2407bb09512f8d497498891e55dee3/lib/chef/knife/ec2_server_create.rb#L1061

This particular line is causing issue where it calls windows_password method before instance is getting created. 

So we should not call above lines of code for Administrator that will automatically resolve the problem for user. 

@btm to answer to your question server gets available after create and when we call when bootstrap_for_window_node we get windows_password since we verify the winrm connection is successful and then call bootstrap https://github.com/MsysTechnologiesllc/knife-ec2/blob/c3371c98ce2407bb09512f8d497498891e55dee3/lib/chef/knife/ec2_server_create.rb#L821

I have checked with until loop until loop goes in infinite loop for @server check since here create_server_def has call to ssl_config_user_data which calls for windows_password method. This call is before we create the server which will always create problem in case of winrm_user as Administrator and if winrm_password is not provided. 

So my suggestion will be just verify do we really need this lines for Administrator user to set user data
https://github.com/MsysTechnologiesllc/knife-ec2/blob/c3371c98ce2407bb09512f8d497498891e55dee3/lib/chef/knife/ec2_server_create.rb#L1061:L1062

Reply:
Adding some more points to @Vasu1105 comment:
For Administrator user and chef on Windows over SSL

While creating server here https://github.com/chef/knife-ec2/blob/master/lib/chef/knife/ec2_server_create.rb#L534 call goes to https://github.com/chef/knife-ec2/blob/master/lib/chef/knife/ec2_server_create.rb#L1179 and subsequently to `ssl_config_user_data` and satisfying all the conditions calls `windows_password` here
https://github.com/chef/knife-ec2/blob/master/lib/chef/knife/ec2_server_create.rb#L1061. But since server is not created yet, password could not be fetched from https://github.com/chef/knife-ec2/blob/master/lib/chef/knife/ec2_server_create.rb#L1540.

So we need to check for the server availability with `if @server`. `until @server` takes this to infinite loop. 
Also in the first turn it goes to the else part here https://github.com/chef/knife-ec2/pull/524/files#diff-e2993b8367db674f58a052036f00052aR1546 and fetches the primary details of server for creating server.

And in second try it satisfy the condition of `if @server` and make the password available from https://github.com/chef/knife-ec2/pull/524/files#diff-e2993b8367db674f58a052036f00052aR1539